csrc/multidevice/communication.cpp

```diff
@@ -328,15 +329,39 @@ c10::intrusive_ptr<c10d::Work> postGather(
       output_tensors, input_tensors, {.rootRank = root_relative_index});
 }
 
+
+std::vector<int64_t> getContiguityPermutation(const at::Tensor& tensor) {
+  auto strides = tensor.strides();
+  std::vector<int64_t> dims(strides.size());
+  std::iota(dims.begin(), dims.end(), 0);
+  std::sort(dims.begin(), dims.end(), [&](int64_t a, int64_t b) {
+    return strides.at(a) > strides.at(b);
+  });
+  return dims;
+}
+
 c10::intrusive_ptr<c10d::Work> postAllgather(
     Communication* communication,
     DeviceIdxType my_device_index,
     c10d::Backend* backend,
     at::Tensor input_tensor,
     at::Tensor output_tensor) {
+  
+  auto sharded_axis = getShardedLogicalAxis(communication->in(), ParallelType::DIDx);
```

Comment:
I am hardcoding `ParallelType::DIDx` for now. Input and output tensorviews may have multiple parallel types of which a particular parallel type is involved in all gather. Can we store that information or the sharding axis in the `Communication` object.
Is there another way to fetch this information without any analysis within `postAllGather`?